src/convex/ConvexAvatarMultiToken.sol

```diff
@@ -0,0 +1,698 @@
+            revert InvalidBps(_minOutBpsCrvToWeth);
+        }
+
+        uint256 minOutBpsCrvToWethMin = minOutBpsCrvToWeth.min;
+        if (_minOutBpsCrvToWeth < minOutBpsCrvToWethMin) {
+            revert LessThanBpsMin(_minOutBpsCrvToWeth, minOutBpsCrvToWethMin);
+        }
+
+        uint256 oldMinOutBpsCrvToWethVal = minOutBpsCrvToWeth.val;
+        minOutBpsCrvToWeth.val = _minOutBpsCrvToWeth;
+
+        emit MinOutBpsCrvToWethValUpdated(_minOutBpsCrvToWeth, oldMinOutBpsCrvToWethVal);
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    // PUBLIC: Owner
+    ////////////////////////////////////////////////////////////////////////////
+
+    function createPrivateVault(uint256 _pid) external onlyOwner {
+        _createPrivateVault(_pid);
+    }
+
+    /// @notice Takes a given amount of asset from the owner and stakes them on private vault. Can only be called by owner.
+    /// @param _pid Pid target to stake into its appropiate vault
+    /// @param _amountAsset Amount of asset to be lock.
+    function depositInPrivateVault(uint256 _pid, uint256 _amountAsset) external onlyOwner returns (bytes32 kekId) {
+        if (_amountAsset == 0) {
+            revert NothingToDeposit();
+        }
+
+        address vaultAddr = privateVaults[_pid];
+        if (vaultAddr == address(0)) {
+            revert NoPrivateVaultForPid(_pid);
+        }
+
+        IStakingProxy proxy = IStakingProxy(vaultAddr);
+        address stakingToken = proxy.stakingToken();
+        IFraxUnifiedFarm farm = IFraxUnifiedFarm(proxy.stakingAddress());
+
+        IERC20MetadataUpgradeable(stakingToken).safeTransferFrom(msg.sender, address(this), _amountAsset);
+        /// NOTE: we always try to lock for the min duration allowed
+        kekId = proxy.stakeLocked(_amountAsset, farm.lock_time_min());
+        /// @dev detailed required to enable withdrawls later
+        kekIds[vaultAddr].push(kekId);
+
+        emit Deposit(stakingToken, _amountAsset, block.timestamp);
+    }
+
+    /// @notice Unstakes all staked assets and transfers them back to owner. Can only be called by owner.
+    /// It will loop thru all existent keks and withdraw fully from each of them.
+    /// @dev This function doesn't claim any rewards.
+    /// @param _pid Pid target to withdraw from avatar private vault
+    function withdrawFromPrivateVault(uint256 _pid) external onlyOwner {
+        address vaultAddr = privateVaults[_pid];
+
+        if (vaultAddr == address(0)) {
+            revert NoPrivateVaultForPid(_pid);
+        }
+
+        IStakingProxy proxy = IStakingProxy(vaultAddr);
+        uint256 lockedBal = IFraxUnifiedFarm(proxy.stakingAddress()).lockedLiquidityOf(vaultAddr);
+        if (lockedBal == 0) {
+            revert NothingToWithdraw();
+        }
+
+        bytes32[] memory keks = kekIds[vaultAddr];
+
+        for (uint256 i = 0; i < keks.length;) {
+            proxy.withdrawLockedAndUnwrap(keks[i]);
```

Comment:
Shouldn't we also add a function to withdraw from a subset of keks since this will fail if not all of them have unlocked?

Reply:
that's a good point, let me think about how to handle perhaps in the loop all those which are unlocked and another method to pinpoint to specific keks if required

Reply:
resolved in 3aa143db86fd36de76d5c5b0f57272d256e6207f